### ISSUE TITLE
EKS cluster failed to Cluster IP for Cluster Domain

### ISSUE THREAD

**Issue by izaac** [rancher/rio#631]
**Describe the bug**
Cluster IP is not set when installing rio in Amazon EKS cluster

**To Reproduce**
1. Create an Amazon EKS cluster
2. Install rio`v0.6.0-alpha.1` in this cluster using the KUBECONFIG.
3. Wait until all the components are finished creating and are Running

`rio info` shows the Cluster Domain is created but not the Cluster IP.

```bash
Rio Version: v0.6.0-alpha.1 (7678d67d)
Rio CLI Version: v0.6.0-alpha.1 (7678d67d)
Cluster Domain: xxxxx.on-rio.io
Cluster Domain IPs:
System Namespace: rio-system
System Ready State: false
Wildcard certificates: true

System Components:
```

Services are also properly created but ENDPOINTs return `404`

**Expected behavior**

Custer IP assigned for the Cluster Domain and ENDPOINTs work

**Kubernetes version & type (GKE, on-prem)**: `kubectl version`
```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.6", GitCommit:"96fac5cd13a5dc064f7d9f4f23030a6aeface6cc", GitTreeState:"clean", BuildDate:"2019-08-19T11:13:49Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.6-eks-5047ed", GitCommit:"5047edce664593832e9b889e447ac75ab104f527", GitTreeState:"clean", BuildDate:"2019-08-21T22:32:40Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```
Type:
**Rio version**: `rio info`
```
Rio Version: v0.6.0-alpha.1 (7678d67d)
```
**Additional context**
`rio systemlogs`
output:
```bash
rio-controller | time="2019-10-28T23:07:05Z" level=info msg="Starting rio-controller, version: v0.6.0-alpha.1, git commit: 7678d67d"
rio-controller | time="2019-10-28T23:07:05Z" level=info msg="Updating CRD services.rio.cattle.io"
rio-controller | time="2019-10-28T23:07:05Z" level=info msg="Updating CRD stacks.rio.cattle.io"
rio-controller | I1028 23:07:05.974564       1 leaderelection.go:235] attempting to acquire leader lease  rio-system/rio...
rio-controller | time="2019-10-28T23:07:05Z" level=info msg="listening at :443"
rio-controller | I1028 23:07:05.987209       1 leaderelection.go:245] successfully acquired lease rio-system/rio
rio-controller | time="2019-10-28T23:07:07Z" level=info msg="Starting /v1, Kind=ConfigMap controller"
rio-controller | time="2019-10-28T23:07:11Z" level=info msg="Starting apps/v1, Kind=Deployment controller"
rio-controller | time="2019-10-28T23:07:11Z" level=info msg="Starting apps/v1, Kind=DaemonSet controller"
rio-controller | time="2019-10-28T23:07:11Z" level=info msg="Starting apps/v1, Kind=StatefulSet controller"
rio-controller | time="2019-10-28T23:07:12Z" level=info msg="Starting tekton.dev/v1alpha1, Kind=TaskRun controller"
rio-controller | time="2019-10-28T23:07:12Z" level=info msg="Starting cert-manager.io/v1alpha2, Kind=Certificate controller"
rio-controller | time="2019-10-28T23:07:13Z" level=info msg="Starting /v1, Kind=Secret controller"
rio-controller | time="2019-10-28T23:07:13Z" level=info msg="Starting /v1, Kind=Service controller"
rio-controller | time="2019-10-28T23:07:13Z" level=info msg="Starting /v1, Kind=Endpoints controller"
rio-controller | time="2019-10-28T23:07:13Z" level=info msg="Starting extensions/v1beta1, Kind=Ingress controller"
rio-controller | time="2019-10-28T23:07:13Z" level=info msg="Starting admin.rio.cattle.io/v1, Kind=PublicDomain controller"
rio-controller | time="2019-10-28T23:07:14Z" level=info msg="Starting admin.rio.cattle.io/v1, Kind=ClusterDomain controller"
rio-controller | time="2019-10-28T23:07:14Z" level=info msg="Starting rbac.authorization.k8s.io/v1, Kind=ClusterRoleBinding controller"
rio-controller | time="2019-10-28T23:07:14Z" level=info msg="Starting rbac.authorization.k8s.io/v1, Kind=ClusterRole controller"
rio-controller | time="2019-10-28T23:07:15Z" level=info msg="Starting rio.cattle.io/v1, Kind=ExternalService controller"
rio-controller | time="2019-10-28T23:07:15Z" level=info msg="Starting rio.cattle.io/v1, Kind=Router controller"
rio-controller | time="2019-10-28T23:07:15Z" level=info msg="Starting rio.cattle.io/v1, Kind=Service controller"
rio-controller | time="2019-10-28T23:07:15Z" level=info msg="Starting rio.cattle.io/v1, Kind=Stack controller"
rio-controller | time="2019-10-28T23:07:16Z" level=info msg="Starting gitwatcher.cattle.io/v1, Kind=GitCommit controller"
rio-controller | time="2019-10-28T23:07:16Z" level=info msg="Starting gloo.solo.io/v1, Kind=Settings controller"
rio-controller | W1028 23:16:51.141761       1 reflector.go:302] github.com/solo-io/gloo/projects/gloo/pkg/api/v1/kube/client/informers/externalversions/factory.go:117: watch of *v1.Settings ended with: too old resource version: 1453 (4663)
rio-controller | W1028 23:21:24.140401       1 reflector.go:302] k8s.io/client-go/informers/factory.go:133: watch of *v1.Role ended with: too old resource version: 1732 (5522)
rio-controller | W1028 23:23:04.664484       1 reflector.go:302] k8s.io/client-go/informers/factory.go:133: watch of *v1beta1.Ingress ended with: too old resource version: 523 (5831)
rio-controller | W1028 23:25:40.986968       1 reflector.go:302] github.com/tektoncd/pipeline/pkg/client/informers/externalversions/factory.go:117: watch of *v1alpha1.TaskRun ended with: too old resource version: 1500 (6312)
rio-controller | W1028 23:26:37.040824       1 reflector.go:302] github.com/rancher/rio/pkg/generated/informers/externalversions/factory.go:118: watch of *v1.Service ended with: too old resource version: 3592 (6483)
rio-controller | W1028 23:28:54.099670       1 reflector.go:302] k8s.io/client-go/informers/factory.go:133: watch of *v1.Secret ended with: too old resource version: 4076 (6906)
rio-controller | W1028 23:29:01.089287       1 reflector.go:302] github.com/rancher/rio/pkg/generated/informers/externalversions/factory.go:118: watch of *v1.Stack ended with: too old resource version: 1503 (6928)
rio-controller | W1028 23:43:54.145526       1 reflector.go:302] k8s.io/client-go/informers/factory.go:133: watch of *v1.ClusterRoleBinding ended with: too old resource version: 1721 (9673)
rio-controller | W1028 23:44:54.646880       1 reflector.go:302] k8s.io/client-go/informers/factory.go:133: watch of *v1.Role ended with: too old resource version: 5522 (9860)
rio-controller | W1028 23:47:31.005583       1 reflector.go:302] github.com/rancher/rio/pkg/generated/informers/externalversions/factory.go:118: watch of *v1.Router ended with: too old resource version: 1503 (10340)
rio-controller | W1029 00:01:27.048931       1 reflector.go:302] github.com/rancher/rio/pkg/generated/informers/externalversions/factory.go:118: watch of *v1.RioInfo ended with: too old resource version: 1257 (12912)
rio-controller | W1029 00:12:04.058616       1 reflector.go:302] k8s.io/client-go/informers/factory.go:133: watch of *v1.RoleBinding ended with: too old resource version: 1733 (14876)
rio-controller | W1029 00:20:14.624327       1 reflector.go:302] k8s.io/client-go/informers/factory.go:133: watch of *v1.RoleBinding ended with: too old resource version: 14876 (16379)
```

`k describe pods gateway-v2-5fdf4545bc-r9x2c`
```bash
Events:
  Type     Reason                  Age                     From                                                  Message
  ----     ------                  ----                    ----                                                  -------
  Normal   Scheduled               6m47s                   default-scheduler                                     Successfully assigned rio-system/gateway-v2-5fdf4545bc-r9x2c to ip-172-31-42-180.us-west-2.compute.internal
  Warning  FailedCreatePodSandBox  6m45s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "7b70db5392b7095793f5031109ed4e299f63d37699f4dc34aeda63be47309d8f" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  6m44s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "1cab260371bb1847e966bbba5feb469784e7fedcafe3e6006d5ccc63e71c6f3a" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  6m43s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "6ee57e4170f2e56689b6ba3be3e8f52f3dd283fce69cb5edd53f8e86f5df258d" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  6m41s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "d22ae05a0e17542b5c63050b5c239eea1ffffe44effead46c7276055ac47d558" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  6m39s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "a8b9e17d9c7b0dcdd7a98d479d6e0c98c14df6c1f2e1dae9c13fcc055da63a6b" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  6m37s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "cdde04240ff8266ced4512010c4967cd6d6b0500a7b2d5aa8c5b9fc5d03ba127" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  6m34s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "045511ac03792c12f0ca0ba72de466334368fd924e710232d442a389da6886dd" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  6m32s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "b0a4daef0b54b841c5ad9bef49539212ddad4b0fc107df2be88b9a7d846f6d02" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Warning  FailedCreatePodSandBox  6m30s                   kubelet, ip-172-31-42-180.us-west-2.compute.internal  Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "0979e5e3bd580af0c3e562584fd17bbd8f15bb5184520cd578489e2d124cb3f0" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
  Normal   SandboxChanged          6m27s (x12 over 6m45s)  kubelet, ip-172-31-42-180.us-west-2.compute.internal  Pod sandbox changed, it will be killed and re-created.
  Warning  FailedCreatePodSandBox  6m24s (x4 over 6m26s)   kubelet, ip-172-31-42-180.us-west-2.compute.internal  (combined from similar events): Failed create pod sandbox: rpc error: code = Unknown desc = failed to set up sandbox container "7c901fa4073dc4434ca31c42fd108caebb1289753fd974e253d47d7372bafc6a" network for pod "gateway-v2-5fdf4545bc-r9x2c": NetworkPlugin cni failed to set up pod "gateway-v2-5fdf4545bc-r9x2c_rio-system" network: add cmd: failed to assign an IP address to container
```

**Comment by cbron** [comment]
Thanks @izaac for detailed breakdown. Seems like it might be a general EKS bug unrelated to Rio, there is a very long thread about it: https://github.com/aws/amazon-vpc-cni-k8s/issues/59

Can you add info about what nodes and instance-types you used ? We might need to recommend something bigger. Let me know when you come back around to this and I can help look into it and any errors.

**Comment by rancher-max** [comment]
This appears to be fixed in v0.6.0-alpha.2. See: https://github.com/rancher/rio/issues/619#issuecomment-548166807

I'll let @izaac verify again with his cluster sizes before closing. It did initially fail for me with 17 available pods in my cluster (created using Rancher with all defaults: kubernetes version=1.14, instance type=m5.large, desired asg size=1, node volume size=20), but worked with a larger cluster.

**Comment by izaac** [comment]
This is working for me too, with `rio alpha.2` the cluster worker node is m5.large as pointed out by @rancher-max. 

EKS Kubernetes Version:`v1.14.6-eks-5047ed`

The default volume size barely passed for me, we need to document a larger volume size for clusters. Rancher UI defaults to 20.
